Core/src/Surfaces/LineSurface.cpp

```diff
@@ -80,27 +79,20 @@ Acts::Result<Acts::Vector2> Acts::LineSurface::globalToLocal(
     const GeometryContext& gctx, const Vector3& position,
     const Vector3& direction, double tolerance) const {
   using VectorHelpers::perp;
-  const auto& sTransform = transform(gctx);
-  const auto& tMatrix = sTransform.matrix();
-  Vector3 lineDirection(tMatrix(0, 2), tMatrix(1, 2), tMatrix(2, 2));
-  // Bring the global position into the local frame
-  Vector3 loc3Dframe = sTransform.inverse() * position;
-  // construct localPosition with sign*perp(candidate) and z.()
-  Vector2 lposition(perp(loc3Dframe), loc3Dframe.z());
-  Vector3 sCenter(tMatrix(0, 3), tMatrix(1, 3), tMatrix(2, 3));
-  Vector3 decVec(position - sCenter);
-  // assign the right sign
-  double sign = ((lineDirection.cross(direction)).dot(decVec) < 0.) ? -1. : 1.;
```

Comment:
The PR description reads 

> In `globalToLocal` we did not respect the direction which is important for the line surface.

But it did take the direction into account here no? So was there a bug here and the new implementation doesn't have it?